ocaml/idl/datamodel.ml

```diff
@@ -5118,6 +5118,10 @@ module SM = struct
             ~ty:(Set String) "required_cluster_stack"
             "The storage plugin requires that one of these cluster stacks is \
              configured and running."
+        ; field ~lifecycle:[] ~qualifier:DynamicRO
+            ~default_value:(Some (VSet [])) ~ty:(Set String)
+            "supported_image_formats"
+            "Image formats supported by the SR (VHD, RAW, Qcow2, ...)"
```

Comment:
Is the capitalisation significant? Would be good to list the supported formats in the exact syntax. The … leaves some imagination.

Reply:
> That said, I'm not sure we want to enfoce any format in XAPI, since someone could implement other formats such as vhdx, vmdk, or anything else.

Andriy spent quite a lot of time to be able to handle qcow2 in the toolstack. I don't think it's going to be possible to support all those image formats without intervention in xapi, and hence why it makes sense to code them in xapi. Then it can act as a gatekeeper, ensuring all allowed formats are properly supported

Reply:
I agree that those image formats most probably won't  be supported without the help of XAPI. So I understand that it makes sense to list the supported format at the XAPI level. I will add the list of supported backends.